tinlake-ui/components/PoolLink/index.tsx

```diff
@@ -1,75 +1,59 @@
 import Link from 'next/link'
 import { useRouter } from 'next/router'
+import React from 'react'
 import { UrlObject } from 'url'
-import config from '../../config'
 
 interface Props {
   href: string | UrlObject
+  children: React.ReactNode
   as?: string | UrlObject
 }
 
 // PoolLink allows navigation within the same pool (it pre-fixes the passed href by the root address)
-export const PoolLink: React.FunctionComponent<Props> = ({ href, as, children }) => {
-  const { root } = useRouter().query
+export const PoolLink: React.FC<Props> = ({ href, as, children }) => {
+  const router = useRouter()
 
-  if (!root) {
-    throw new Error('expected `root` to be in route query, but was not')
-  }
-
-  // if set, require as to be string if href is string, and require it to be UrlObject if as is UrlObject
   if (as !== undefined && typeof href !== typeof as) {
     throw new Error('as and href need to be the same type')
   }
 
   let poolHref: string | UrlObject = ''
   let poolAs: string | UrlObject = ''
   if (typeof href === 'string') {
-    poolHref = getHref(root, href)
-    poolAs = getAs(root, as || href)
+    poolHref = getHref(router.query.root, href)
+    poolAs = getAs(router.query.root, router.query.slug, as || href)
   } else {
     poolHref = {
       ...href,
-      pathname: getHref(root, href.pathname),
+      pathname: getHref(router.query.root, href.pathname),
     }
     poolAs = {
       ...((as as UrlObject) || href),
-      pathname: getAs(root, href.pathname),
+      pathname: getAs(router.query.root, router.query.slug, href.pathname),
     }
   }
+
   return (
     <Link href={poolHref} as={poolAs} shallow>
       {children}
     </Link>
   )
 }
 
-function getHref(rootOrSlug: string | string[], href: string | null | undefined | UrlObject): string {
-  console.log("GETHREF", config.pools, config.upcomingPools)
-  const pool = config.pools.find(
-    (p) => (rootOrSlug as string).toLowerCase() === p.addresses.ROOT_CONTRACT.toLowerCase()
-  )
-  if (pool) {
+function getHref(root: string | string[] | undefined, href: string | null | undefined | UrlObject): string {
+  if (root) {
     return `/pool/[root]/[slug]${href}`
   }
-  const upPool = config.upcomingPools.find((p) => (rootOrSlug as string) === p.metadata.slug)
-  if (upPool) {
-    return `/pool/[root]${href}`
-  }
-
-  throw new Error(`could not find root ${rootOrSlug} for href in pools or upcoming pools`)
+  return `/pool/[slug]${href}`
 }
 
-function getAs(rootOrSlug: string | string[], as: string | null | undefined | UrlObject): string {
-  const pool = config.pools.find(
-    (p) => (rootOrSlug as string).toLowerCase() === p.addresses.ROOT_CONTRACT.toLowerCase()
-  )
-  if (pool) {
-    return `/pool/${rootOrSlug}/${pool.metadata.slug}${as}`
+function getAs(
+  root: string | string[] | undefined,
+  slug: string | string[] | undefined,
+  as: string | null | undefined | UrlObject
+): string {
+  if (root) {
```

Comment:
This should test for `slug` again, right? 

```suggestion
  if (slug) {
```